obspy/io/nordic/core.py

```diff
@@ -164,20 +164,40 @@ def _readheader(head_lines):
     # Construct a rough catalog, then merge events together to cope with
     # multiple origins
     _cat = Catalog()
+    # Make a list of the line numbers that are associated with proper origins
+    # (i.e., not just extended origin lines)
+    origin_line_numbers = []
     for line in head_lines:
         _cat.append(_read_origin(line=line[0]))
+        origin_line_numbers.append(line[1])
     new_event = _cat.events.pop(0)
-    for event in _cat:
+
+    for event, orig_line_number in zip(_cat, origin_line_numbers[1:]):
```

Comment:
Might be good to use a pop statement on the origin line list above and avoid that index here in the loop header, to be safe from future errors with unaligned lists.

Reply:
Fixed that; anyways I wasn't handling the list well as I was removing items in one of the lists that was being looped through.